File: demo_app/app/src/main/java/com/bandyer/demo_android_sdk/adapter_items/SelectedUserItem.java
package com.bandyer.demo_android_sdk.adapter_items;

import android.view.View;

import androidx.annotation.NonNull;

import com.bandyer.demo_android_sdk.R;
import com.google.android.material.chip.Chip;
import com.mikepenz.fastadapter.items.AbstractItem;

import java.util.List;

import butterknife.BindView;

/**
 * A simple RecyclerView item used to display the user userAlias as a Chip in the list.
 */
public class SelectedUserItem extends AbstractItem<SelectedUserItem, SelectedUserItem.ViewHolder> {

    public final String userAlias;
    public final int position;

    public SelectedUserItem(String userAlias, int position) {
        this.userAlias = userAlias;
        this.position = position;
    }

    @Override
    public int getType() {
        return R.id.user_item_id;
    }

    @Override
    public int getLayoutRes() {
        return R.layout.selected_user_item;
    }

    @NonNull
    @Override
    public SelectedUserItem.ViewHolder getViewHolder(@NonNull View v) {
        return new SelectedUserItem.ViewHolder(v);
    }

    protected static class ViewHolder extends ButterKnifeViewHolder<SelectedUserItem> {

        @BindView(R.id.userAlias)
        Chip userAlias;

        ViewHolder(View view) {
            super(view);
            view.setFocusable(false);
            view.setFocusableInTouchMode(false);
        }

        @Override
        public void bindView(@NonNull SelectedUserItem item, @NonNull List<Object> payloads) {
            userAlias.setText(item.userAlias);
            userAlias.setOnCloseIconClickListener(v -> userAlias.performClick());
        }

        @Override
        public void unbindView(@NonNull SelectedUserItem item) {
            userAlias.setText(null);
        }
    }
}

File: app_utilities/src/main/java/com/bandyer/app_utilities/adapter_items/SelectedUserItem.kt
/*
 * Copyright (C) 2020 Bandyer S.r.l. All Rights Reserved.
 * See LICENSE.txt for licensing information
 */

package com.bandyer.app_utilities.adapter_items

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.bandyer.app_utilities.R
import com.mikepenz.fastadapter.items.AbstractItem
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.selected_user_item_layout.view.*

/**
 * A simple RecyclerView item used to display the user userAlias as a Chip in the list.
 */
class SelectedUserItem(@JvmField val userAlias: String,@JvmField val position: Int) : AbstractItem<SelectedUserItem.ViewHolder>() {

    override var identifier: Long = position.toLong()
    override val type: Int = R.id.user_selected_item
    override val layoutRes: Int = R.layout.selected_user_item_layout
    override fun getViewHolder(v: View): ViewHolder = ViewHolder(v)

    override fun bindView(holder: ViewHolder, payloads: List<Any>) {
        super.bindView(holder, payloads)
        holder.containerView.userAlias!!.text = userAlias
        holder.containerView.userAlias!!.setOnCloseIconClickListener { v: View? -> v!!.performClick() }
    }

    override fun unbindView(holder: ViewHolder) {
        super.unbindView(holder)
        holder.containerView.userAlias.text = null
    }

    class ViewHolder(val containerView: View) : RecyclerView.ViewHolder(containerView) {
        init {
            containerView.isFocusable = false
            containerView.isFocusableInTouchMode = false
        }
    }

}